test/e2e_test.go

```diff
@@ -539,17 +581,33 @@ func watchPodReadyStatus(client clientset.Interface, podNamespace string, podNam
 			if !ok {
 				return fmt.Errorf("Watch pod failed")
 			}
-			var containerReady = false
+			var (
+				containerReady     = false
+				initContainerReady = false
+			)
 			if pod.Name == podName {
 				for _, containerStatus := range pod.Status.ContainerStatuses {
 					if !containerStatus.Ready {
 						break
 					}
 					containerReady = true
 				}
-				if containerReady {
+
+				if containerReady && podName != initSidecarContainersPodName {
 					return nil
 				}
+				if podName == initSidecarContainersPodName {
```

Comment:
I think you can simply iterate both collections instead of checking for specific pod name. All init containers must be `Ready=true` if they run to completion yet and if they are sidecars they will follow the same logic as a regular containers

Reply:
thanks, I have updated it.
could you please take a look again?